File: 1800-maximum-ascending-subarray-sum/1800-maximum-ascending-subarray-sum.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxAscendingSum = function(nums) {
    let res = nums[0]
    let cur = nums[0]
    for (let i=1; i < nums.length; i++) {
        if (nums[i-1] < nums[i]) {
            cur += nums[i]
        } else {
            cur = nums[i]
        }
        res = Math.max(res, cur)
    }
    return res
};

File: 1800-maximum-ascending-subarray-sum/1800-maximum-ascending-subarray-sum.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        max_sum = cur_sum = nums[0]
        for i in range(1, len(nums)):
            if nums[i-1] < nums[i]:
                cur_sum += nums[i]
            else:
                cur_sum = nums[i]
            max_sum = max(max_sum, cur_sum)
        return max_sum